File: engine/engine-model/src/main/scala/net/server/channel/packet/interaction/AddItemPlayerInteractionPacket.scala
package net.server.channel.packet.interaction

class AddItemPlayerInteractionPacket(private var _mode: Byte, private var _slotType: Byte, private var _slot: Short, private var _bundles: Short, private var _perBundle: Short, private var _price: Int) extends BasePlayerInteractionPacket(_mode = _mode) {
  def slotType: Byte = _slotType

  def slot: Short = _slot

  def bundles: Short = _bundles

  def perBundle: Short = _perBundle

  def price: Int = _price
}

File: engine/engine-model/src/main/java/net/server/channel/packet/interaction/AddItemPlayerInteractionPacket.java
package net.server.channel.packet.interaction;

public class AddItemPlayerInteractionPacket extends BasePlayerInteractionPacket {
   private final Byte slotType;

   private final Short slot;

   private final Short bundles;

   private final Short perBundle;

   private final Integer price;

   public AddItemPlayerInteractionPacket(Byte mode, Byte slotType, Short slot, Short bundles, Short perBundle, Integer price) {
      super(mode);
      this.slotType = slotType;
      this.slot = slot;
      this.bundles = bundles;
      this.perBundle = perBundle;
      this.price = price;
   }

   public Byte slotType() {
      return slotType;
   }

   public Short slot() {
      return slot;
   }

   public Short bundles() {
      return bundles;
   }

   public Short perBundle() {
      return perBundle;
   }

   public Integer price() {
      return price;
   }
}
